### ISSUE TITLE
Problem updating email with an existing address

### ISSUE THREAD

**Issue by santanaruben** [jon-sully/react-netlify-identity-gotrue#4]
If I try to update the email with an address that already exists, the `identity.user` data is updated with an error message 

`ni.user:"{"code":422, "msg": "Email address already registered by another user"}"`

 and the token is unusable. Then I must either log out or perform a localStorage cleanup. In any case, the application generates an error because many variables depend on the `identity.user` data.

**Comment by jon-sully** [comment]
Uh oh! That sounds like a problem. Let me check it out and add the appropriate error-handling code 🙂

**Comment by jon-sully** [comment]
This has been fixed and is live in v0.3.5. The `.update()` method now throws an error with the GoTrue error message if GoTrue sends back an error. Given that, I've refactored my demo user form submit action to look more like:

```js
  const onSubmit = async (data) => {
    setFormProcessing(true)

    await identity.update(data)
      .then(_ => {
        setValue('newPasswordOne', '')
        setValue('newPasswordTwo', '')
        setFormProcessing(false)
      })
      .catch(e => {
        setFormProcessing(false)
        setFormError(e.message)
        return
    })
  }
```

Leveraging the `.then()` with `.catch()` workflow 👍🏻
